sensors/nxt_i2c_sensor_defs.c

```diff
@@ -262,6 +262,19 @@ static int ht_angle_scale(void *context,
 	return 0;
 }
 
+static int ms_pps58_nx_set_mode_post_cb(nxt_i2c_sensor_data *sensor, u8 mode)
+{
+	struct i2c_client *client = sensor->client;
+	int ref;
+
+	if(mode < 4 || 6 < mode)
+		return 0;
+
+	ref = i2c_smbus_read_word_data(client, 0x43);
+	if(ref < 0)
+		return ref;
+	return i2c_smbus_write_word_data(client, 0x47, ref);
```

Comment:
Setting to the current value here would have the same effect as the ZERO command, would it not? If the sensor is already under pressure, this could be undesirable. I was thinking that we should set this to standard atmospheric pressure depending on the units.

    switch (mode) {
    case 4:
        value = 15; /* PSI */
        break;
    case 5:
        value = 1013; /* mbar */
        break;
    case 6:
        value = 101; /* kPa */
        break;
    default:
        return 0;
    }

    return i2c_smbus_write_word_data(client, 0x47, value)